### DISCUSSION TITLE
gas efficient

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#2566] [Q&A] [answered]
GN. i remember someone wrote a writeup and told the function which is `payable` cost lower gas than a `non payable` function. so, why we don't make all of the functions `payable` ?

**Comment by EngrPips** [comment, marked as answer]
Yeah, if you mark a function as payable, it helps to reduce the gas cost because there won't be a zero value check when you call the function, but not marking it as payable will include the zero value check whenever you call that function and revert if any value is sent. That is a kind of safeguard put in place in case you mistakenly send ether along with a transaction to a contract that can't handle the ether. Now the gas expense when you don't mark the function as payable isn't that too much that we are willing to unknowingly lock our ether in a smart contract forever.

**Reply by MostPow3rful** [reply]
interesting. Thanks for your good explain

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
